### PR TITLE
Complete fixes for various failures in self-test suite

### PR DESCRIPTION
**Pull request by anoopcs9**

### CODE REVIEW

File: src/glfs-cli.c [hunk 1]
```diff
@@ -271,7 +271,7 @@ parse_options (struct cli_context *ctx)
         opterr = 0;
 
         while (true) {
-                opt = getopt_long (argc, argv, "o:", long_options,
```
Comment (craigcabrey) [inline src/glfs-cli.c:274]: The main reason I left out `-h` as a short option for help text was to mimic coreutils which does not have a short option for help.

Additionally, I wanted to keep consistency between the tools. Specifically, the `gfls` tool has a `-h` flag which means human readable.

Comment (anoopcs9) [inline src/glfs-cli.c:274]: > The main reason I left out -h as a short option for help text was to mimic coreutils which does not have a short option for help.

Does that mean we can get rid of existing -x short option for --help for every core utility? If so I will do the necessary changes and update the commits.

> Additionally, I wanted to keep consistency between the tools. Specifically, the gfls tool has a -h flag which means human readable.

I am not justifying the change but this was the reason for which I didn't add -h for gfls.

Comment (craigcabrey) [inline src/glfs-cli.c:274]: The `'x'` used in the case statement is just used to refer to the long option. If it's not included in the optstring, then `getopt_long` doesn't expose it as a short option at runtime.

``` c
static struct option const long_options[] =
{
        {"help", no_argument, NULL, 'x'},
        {NULL, no_argument, NULL, 0}
}

opt = getopt_long (argc, argv, "", long_options, &option_index);

switch (opt) {
  case 'x': //
}
```

vs.

``` c
static struct option const long_options[] =
{
        {"help", no_argument, NULL, 'x'},
        {NULL, no_argument, NULL, 0}
}

opt = getopt_long (argc, argv, "x", long_options, &option_index);

switch (opt) {
  case 'x': //
}
```

In the former calling `./bin/gf -x` results in an error whereas the latter does not.

Comment (anoopcs9) [inline src/glfs-cli.c:274]: > The 'x' used in the case statement is just used to refer to the long option. If it's not included in the optstring, then getopt_long doesn't expose it as a short option at runtime.

Ok, I understand. So do we really need short help flag tests in .t files?

Comment (craigcabrey) [inline src/glfs-cli.c:274]: We can nuke those.

Comment (anoopcs9) [inline src/glfs-cli.c:274]: Fine. I will remove those and make the changes when you are done with your remaining reviews. Thanks.

**Comment by anoopcs9** [comment]
@craigcabrey I have updated the commits removing the 'short help' flag tests.

**Comment by craigcabrey** [comment]
Apologies on the delay, I've been wrapped up in other projects. Looks good to me.

**Pull request merged**